apps/openmw/mwscript/miscextensions.cpp

```diff
@@ -190,11 +192,11 @@ namespace MWScript
                     {
                         MWBase::Environment::get().getWorld()->activateDoor(ptr, 0);
 
-                        float xr = ptr.getCellRef().getPosition().rot[0];
-                        float yr = ptr.getCellRef().getPosition().rot[1];
-                        float zr = ptr.getCellRef().getPosition().rot[2];
+                        osg::Quat xr = osg::Quat(ptr.getCellRef().getPosition().rot[0], -osg::X_AXIS);
+                        osg::Quat yr = osg::Quat(ptr.getCellRef().getPosition().rot[1], -osg::Y_AXIS);
+                        osg::Quat zr = osg::Quat(ptr.getCellRef().getPosition().rot[2], -osg::Z_AXIS);
 
-                        MWBase::Environment::get().getWorld()->rotateObject(ptr, xr, yr, zr);
+                        MWBase::Environment::get().getWorld()->rotateWorldObject(ptr, zr * yr * xr);
```

Comment:
@elsid the navigator seems to ignore rotations performed through rotateWorldObject, so it may not perform as intended with RotateWorld instructions (it currently ignores script-locked doors with this fix). Any thoughts?

Reply:
That's right. #2492 fixes this.